module/IlluminaBeadArrayFiles.py

```diff
@@ -576,8 +576,8 @@ def normalize_intensities(self, x, y, threshold = True):
         tempx3 = tempx2 - self.shear * tempy2
         tempy3 = tempy2
 
-        xn = tempx3 / self.scale_x
-        yn = tempy3 / self.scale_y
+        xn = tempx3 / self.scale_x if self.scale_x != 0 else nan
+        yn = tempy3 / self.scale_y if self.scale_y != 0 else nan
 
         if threshold:
             xn = max(0, xn)
```

Comment:
Yes, that is a good point. I'll change this so that the NaN survives the thresholding.